### PR TITLE
[TPU] fixed bug for finetuning and pretraining

### PR DESCRIPTION
**Pull request by gkroiz**
There was a `fabric.print()` statement before `fabric.launch()`, which cannot happen on TPUs. The solution was to move `fabric.print()` into the launched function

### CODE REVIEW

**Review by awaelchli — APPROVED** [review]

**Comment by awaelchli** [comment]
Thanks @gkroiz 
I believe we should nonetheless address this in Fabric directly by detecting that we are going to run with TPU multiprocessing. It should be perfectly valid to print in the main process, since no other processes exist.

**Comment by awaelchli** [comment]
I created an issue for this :)
https://github.com/Lightning-AI/lightning/issues/18168

**Comment by carmocca** [comment]
The advantage of printing this before launch is to have it show if it hangs at the very beginning. Lets fix it in lightning directly

**Comment by carmocca** [comment]
Merging to unblock and I'll revert this once it's fixed in Lightning

**Comment by awaelchli** [comment]
@gkroiz Looking at the XLA code, I was wondering whether this should be addressed in XLA directly. We are calling these functions here (e.g. when calling fabric.local_rank) it goes to:

https://github.com/pytorch/xla/blob/498bce41fccac29a1f1a4310ed7779102057cc78/torch_xla/core/xla_model.py#L140

This defines a default value "defval" and for xrt this worked well, but for pjrt this function accesses the device regardless:
https://github.com/pytorch/xla/blob/498bce41fccac29a1f1a4310ed7779102057cc78/torch_xla/runtime.py#L164C63-L164C63
(I believe this is the reason for the error you encountered). 
I'm fine guarding this in Lightning, just wanted to raise the potential inconsistency or unintended change if I'm not mistaken.
